### PR TITLE
Remove service, migrate all to service-new

### PR DESCRIPTION
**Pull request by coriolinus**
Makes progress towards https://github.com/paritytech/polkadot/issues/1620. _Does not_ close that issue.

Integrates #1697.

Note: because this work involves replacing the old `service` with `service-new`, this needs some burn-in to ensure that we can still follow the existing chains and author blocks.

Note: initially, the goal was to do all the work necessary for #1620 in this PR, hence the branch name. However, recently I've been spending a lot of time just keeping up with changes to `master` which update `service` but not `service-new`, and that gets old fast. There are therefore a couple of changes in here not related to the service update, which would probably be more effort than they're worth to extract. 

[edit] Burnin [running](https://github.com/paritytech/devops/wiki/Polkadot-PR-Deployments) on `kusama-sentry-ue1-0` / `kusama-validator-ue1-0`

### CODE REVIEW

File: runtime/test-runtime/src/lib.rs
Comment (rphmeier) [inline runtime/test-runtime/src/lib.rs:555]: Note that there is a runtime_api_impl module in `runtime/parachains` that you should be able to forward calls to.

Comment (rphmeier) [reply runtime/test-runtime/src/lib.rs:555]: You'll need to add all parachains modules to the test runtime for that, though.

Comment (coriolinus) [reply runtime/test-runtime/src/lib.rs]: Do I? It looks like this is the only change that the runtime actually requires:

https://github.com/paritytech/polkadot/blob/73ded05a6359bd2c32c650817a84b82baf2ed55a/runtime/test-runtime/src/lib.rs#L472-L473

Will adding the rest of them help somehow?

File: Cargo.toml
Comment (montekki) [inline Cargo.toml:82]: This doesn't look like this should be removed should it?

File: node/service/src/chain_spec.rs
Comment (montekki) [inline node/service/src/chain_spec.rs:862]: Probably omitting formatting fixes from PRs of this size would make things easier

File: runtime/kusama/src/lib.rs
Comment (montekki) [inline runtime/kusama/src/lib.rs:968]: It seems to me that all relevant paras modules should go in here?

File: Cargo.toml
Comment (coriolinus) [reply Cargo.toml:82]: This is promoting the service rewrite to be the service, which means that as there are no longer two services, we no longer need the feature to choose the rewrite.

File: runtime/kusama/src/lib.rs
Comment (coriolinus) [reply runtime/kusama/src/lib.rs:968]: It's hard for me to judge what the effect of including, or excluding, the various paras modules would actually be.

File: node/service/src/chain_spec.rs
Comment (bkchr) [inline node/service/src/chain_spec.rs:592]: I don't understand why you not copied the old function...

File: node/service/src/chain_spec.rs
Comment (bkchr) [reply node/service/src/chain_spec.rs:862]: That all looks like rustfmt formatting

File: node/service/src/chain_spec.rs
Comment (coriolinus) [reply node/service/src/chain_spec.rs:592]: I'm fairly sure that I copied this from _somewhere_. If there's a better source, I'm happy to update it.

File: node/service/src/chain_spec.rs
Comment (coriolinus) [reply node/service/src/chain_spec.rs:862]: Yes to both, but if we omit formatting for ease of review, then nothing ever gets formatted, which makes things harder over time. I can attempt to revert the formatting to the ad-hoc nature it had before if that seems important, but I'd prefer not to.

File: node/service/src/chain_spec.rs
Comment (bkchr) [reply node/service/src/chain_spec.rs:592]: You did not copy it from Rococo. This is westend or whatever.

File: node/service/Cargo.toml
Comment (ordian) [inline node/service/Cargo.toml:9]: This is not introduced in this PR, but not a fan of package renames like these, unless there is a clear guideline how to do these renames that is followed by everyone, but there isn't one AFAIK.

**Review by ordian — COMMENTED** [review]
The code LGTM. However, I'm hesitant to give an approval, because it's not clear to me what is missing (e.g. https://github.com/paritytech/polkadot/pull/1630#discussion_r488728339), so I'll leave it to someone more knowledgeable.

File: node/service/Cargo.toml
Comment (coriolinus) [reply node/service/Cargo.toml:9]: Agree, I prefer to avoid them also.

File: runtime/kusama/src/lib.rs
Comment (coriolinus) [reply runtime/kusama/src/lib.rs]: Added them in 06e960d.

File: runtime/polkadot/src/lib.rs
Comment (coriolinus) [inline runtime/polkadot/src/lib.rs:129]: I'm guessing that the parachains modules should not be callable, but I'm not certain, and would appreciate :+1: or :-1:  from someone who knows.

File: node/service/src/chain_spec.rs
Comment (coriolinus) [reply node/service/src/chain_spec.rs:592]: Hmm. Not sure where I should have copied it from. The function doesn't exist in [`rococo-branch`](https://github.com/paritytech/polkadot/blob/rococo-branch/node/service/src/chain_spec.rs) or [`bkchr-rococo`](https://github.com/paritytech/polkadot/blob/bkchr-rococo/node/service/src/chain_spec.rs) [`cumulus-branch`](https://github.com/paritytech/polkadot/blob/cumulus-branch/node/service/src/chain_spec.rs).

File: cli/Cargo.toml
Comment (ordian) [inline cli/Cargo.toml:26]: dumb question: with this, it's not possible to opt-out of the `full-node` feature, how would it work with the light client?

File: node/service/src/chain_spec.rs
Comment (bkchr) [reply node/service/src/chain_spec.rs:592]: You should check the correct file: https://github.com/paritytech/polkadot/blob/26f1fa47f7836ab4bee5d4aad127ebce748320dd/service/src/chain_spec.rs#L362

File: node/service/Cargo.toml
Comment (bkchr) [reply node/service/Cargo.toml:9]: And why did you keep them?

File: cli/Cargo.toml
Comment (coriolinus) [reply cli/Cargo.toml:26]: Not dumb. Maybe I need to introduce a `full-node` feature in the CLI.

Comment (bkchr) [reply cli/Cargo.toml:26]: The feature name is misleading. This feature is used to compile the node for the browser, where we only run the light client. But the "native" light client can be run with this feature enabled.

Comment (ordian) [reply cli/Cargo.toml:26]: Maybe it's worth renaming it to `native-client` to differentiate between `native` and `browser` environments.

Comment (coriolinus) [reply cli/Cargo.toml]: Updated in c0e9e437.

Comment (bkchr) [reply cli/Cargo.toml]: @coriolinus did you read my comment above? Propagating the feature makes no sense. Cli will not be used by the browser.

Comment (ordian) [reply cli/Cargo.toml]: @bkchr what is the `browser` feature used for then?

Comment (bkchr) [reply cli/Cargo.toml]: I see, I assumed some different structure.

Comment (bkchr) [reply cli/Cargo.toml]: So, it was before already propagated, but using the `service` feature.

File: runtime/polkadot/src/lib.rs
Comment (bkchr) [inline runtime/polkadot/src/lib.rs:971]: Why did you added all this stuff? This makes this pr "unmergable". 

This will require a lot of testing and are these pallets already ready?

They also change the call encoding in an invasive way.

File: cli/Cargo.toml
Comment (ordian) [reply cli/Cargo.toml]: TBH, I don't know if the `browser` feature of polkadot's cli was ever tested (cc @expenses), AFAIK people experimented with the browser demo directly in substrate.

So it may be out of scope of this PR, we can fix it later.

File: node/service/Cargo.toml
Comment (coriolinus) [reply node/service/Cargo.toml:9]: Kept them because it didn't seem worth chasing them down to replace them wherever in the codebase they happened to be. Reordering the dependencies to make them easier to understand is one thing; renaming them is more work.

File: cli/Cargo.toml
Comment (coriolinus) [reply cli/Cargo.toml]: At this point I am a bit confused. @bkchr No, I didn't see your comment until after I'd added the patch propagating the feature. What's your recommended course of action at this point?

File: runtime/polkadot/src/lib.rs
Comment (coriolinus) [reply runtime/polkadot/src/lib.rs:971]: I added these because of https://github.com/paritytech/polkadot/pull/1630/#discussion_r487049798 and https://github.com/paritytech/polkadot/pull/1630/#discussion_r488728339. If that's the wrong action, it's easy to revert.

I mentioned earlier: I don't have any intuitive understanding of what the consequences of that change will be. The code compiles, which is clearly a necessary but sufficient step. I don't know what the rest of the steps are.

File: cli/Cargo.toml
Comment (bkchr) [reply cli/Cargo.toml]: TLDR your commit is fine

File: node/service/src/chain_spec.rs
Comment (coriolinus) [reply node/service/src/chain_spec.rs]: Updated in 34e51e5e.

File: Cargo.lock
Comment (ordian) [inline Cargo.lock:2782]: this seems to be an accidental downgrade when resolving merge conflicts

Comment (coriolinus) [reply Cargo.lock:2782]: I went through just now and, in every case where `Cargo.lock` showed a downgrade relative to master, manually reset it to the newer version. Then I did `cargo check --all`, and all my edits got reverted. Wherever those new versions were specified--mainly `libp2p-*`--they no longer appear to be.

Comment (ordian) [reply Cargo.lock]: Pushed cf2062a. I just did `git checkout origin/master Cargo.lock && cargo check --workspace`. Hopefully, this will fix it.

Comment (coriolinus) [reply Cargo.lock]: That commit causes test failures. I don't know _why_ it does--the mechanism of the failure isn't obvious--but it does seem to.

Comment (ordian) [reply Cargo.lock]: Hmm, I think a merge with master might resolve it, let me try.

File: runtime/polkadot/src/lib.rs
Comment (rphmeier) [reply runtime/polkadot/src/lib.rs:971]: I also don't believe we should yet alter the Polkadot or Kusama runtime module sets, if possible. It would make most sense to me to provide a dummy implementation of the `v1::ParachainHost` in each of those runtimes, but other than that not to alter them.

Comment (bkchr) [reply runtime/polkadot/src/lib.rs:971]: Just merge master to get these dummy implementations of the runtime apis.

Comment (coriolinus) [reply runtime/polkadot/src/lib.rs]: Done in df7a758... but something's up with CI, apparently; the checks don't appear to be running.

**Review by ordian — APPROVED** [review]
LGTM

**Comment by bkchr** [comment]
Was this also tested on Polkadot nodes?

We can not merge a pr that touches such a sensible part of the code base without testing it on Polkadot.

**Comment by tomaka** [comment]
Nothing problematic, but on the burnin I see that we're running 6 tasks named `DummySubsystem`, is that normal?

**Comment by coriolinus** [comment]
Yes, for now. Replacing those with real subsystems is next up.

**Comment by bkchr** [comment]
> Nothing problematic, but on the burnin I see that we're running 15 tasks/node named `DummySubsystem`, is that normal?

We still need to test it at least once on Polkadot.

File: node/service/src/grandpa_support.rs
Comment (gavofyork) [inline node/service/src/grandpa_support.rs:281]: was this done intentionally or is an automatic formatter being used?

File: node/service/src/grandpa_support.rs
Comment (gavofyork) [inline node/service/src/grandpa_support.rs:286]: with the exception of 1-item tuples, `,)` is an invalid string in our codebase.

File: cli/src/command.rs
Comment (gavofyork) [inline cli/src/command.rs:188]: invalid formatting (though this PR makes it no worse).

File: node/service/src/grandpa_support.rs
Comment (gavofyork) [inline node/service/src/grandpa_support.rs:63]: previous formatting was better.

File: node/service/src/grandpa_support.rs
Comment (coriolinus) [reply node/service/src/grandpa_support.rs:286]: OK: 372f4f6a.

File: node/service/src/grandpa_support.rs
Comment (coriolinus) [reply node/service/src/grandpa_support.rs:281]: In this case rustfmt generated the change.

File: node/service/src/grandpa_support.rs
Comment (coriolinus) [reply node/service/src/grandpa_support.rs:63]: OK: b7b45b3e.

File: cli/src/command.rs
Comment (coriolinus) [reply cli/src/command.rs:188]: OK: 944b745e.

File: node/service/src/grandpa_support.rs
Comment (coriolinus) [reply node/service/src/grandpa_support.rs:286]: Though I suspect that it is also valid when used appropriately in macros, i.e. https://github.com/paritytech/polkadot/blob/c68aee352b84321b6a5691d38e20550577d60a45/node/network/availability-distribution/src/tests.rs#L35

**Pull request merged**
